util/network-devp2p/src/discovery.rs

```diff
@@ -115,13 +128,14 @@ pub struct Discovery<'a> {
 	id_hash: H256,
 	secret: Secret,
 	public_endpoint: NodeEndpoint,
+	started_discovering: bool,
+	discovering: bool,
```

Comment:
So `started_discovery` is used in order to start the first round of discovery _ever_. The idea is that it is set to `false` at the beginning, then we add some peers (bootnodes), ping them, and as soon as the pings are answered or timedout, we start the first discovery round. Then we start a new round every 10s or 30s depending on the number of peers. Without this, we would have to wait for an extra 10s for it to run for the first time.
The `discovering` bool is used to prevent starting a new discovery round while there is already one going.

Reply:
I see, much clearer now. Can't really think of a better name, maybe s/`started_discovering`/`discovery_initiated`/. Or just a comment here?

Reply:
Sounds better, thanks!